app/controllers/services_controller.rb

```diff
@@ -11,14 +11,66 @@ class ServicesController < ApplicationController
 
   def index
     if params[:category_id]
-      all_services = find_by_category(params[:category_id])
+      all_services = find_by_categories(params[:category_id], params[:eligibility_id])
     elsif params[:eligibility_id]
       all_services = find_by_eligibility(params[:eligibility_id])
     end
 
     render json: ServicesWithResourcePresenter.present(all_services)
   end
 
+  # rubocop:disable AbcSize
+  # rubocop:disable MethodLength
+  # rubocop:disable PerceivedComplexity
+  def find_by_categories(categories_id_string, eligibilities_id_string)
+    query_string = ""
+    if categories_id_string.include? ","
+      category_ids = categories_id_string.split ","
+      first_cat = true
+      category_ids.each do |cat_id|
+        if first_cat
+          query_string = "services.id in (select service_id from categories_services where category_id=" + cat_id + ")"
+          first_cat = false
+        else
+          query_string = query_string +
+                         " and services.id in (select service_id from categories_services where category_id=" + cat_id + ")"
+        end
```

Comment:
I think what you're trying to do might be simpler using an approach like this: https://stackoverflow.com/a/15977167

I think that'd avoid having to dynamically generate a query with an arbitrary number of expressions, since you can push everything into that inner join with a `HAVING` clause. Man, I haven't used the `HAVING` keyword in forever...

Reply:
Not sure how that would make things easier. I'll contact you to get more information.